src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.biomesAPI.Biome;
+import org.terasology.biomesAPI.OnBiomeChangedEvent;
+import org.terasology.engine.Time;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnChangedComponent;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.characters.CharacterMoveInputEvent;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.health.RegenComponent;
+import org.terasology.logic.health.event.ActivateRegenEvent;
+import org.terasology.logic.health.event.DeactivateRegenEvent;
+import org.terasology.logic.health.event.DoDamageEvent;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.particles.components.generators.VelocityRangeGeneratorComponent;
+import org.terasology.physics.events.ImpulseEvent;
+import org.terasology.physics.events.MovedEvent;
+import org.terasology.registry.In;
+/*
+import org.terasology.thirst.ThirstUtils;
+import org.terasology.thirst.component.ThirstComponent;
+import org.terasology.utilities.random.Random;
+
+import static org.terasology.logic.health.RegenAuthoritySystem.BASE_REGEN;
+import static org.terasology.thirst.ThirstAuthoritySystem.THIRST_DAMAGE_ACTION_ID;
+*/
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ExtremeClimateConditionSystem extends BaseComponentSystem{
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private EntityManager entityManager;
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Time time;
+
+    private int healthDecreaseInterval = 20000;
+    private int healthDecreaseAmount= 10;
+    private float thresholdHeight= (float)60;
```

Comment:
Instead of casting the integer value to a floating point you can specify it by appending an `f` to the number:
```suggestion
    private float thresholdHeight= 60f;
```

See https://en.wikibooks.org/wiki/Java_Programming/Primitive_Types for more on primitive types.